src/librustc_mir/interpret/eval_context.rs

```diff
@@ -190,8 +189,11 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> LayoutOf for InterpCx<'mir, 'tcx, M> {
 
     #[inline]
     fn layout_of(&self, ty: Ty<'tcx>) -> Self::TyLayout {
-        self.tcx.layout_of(self.param_env.and(ty))
-            .map_err(|layout| InterpError::Layout(layout).into())
+        self.tcx
+            .layout_of(self.param_env.and(ty))
+            .map_err(|layout| {
+                err_inval!(Layout(layout)).into()
```

Comment:
I think this into can be removed

Reply:
```
error[E0308]: mismatched types
   --> src/librustc_mir/interpret/eval_context.rs:192:9
    |
191 |       fn layout_of(&self, ty: Ty<'tcx>) -> Self::TyLayout {
    |                                            -------------- expected `std::result::Result<rustc_target::abi::TyLayout<'_, &rustc::ty::TyS<'_>>, rustc::mir::interpret::InterpErrorInfo<'_>>` because of return type
192 | /         self.tcx
193 | |             .layout_of(self.param_env.and(ty))
194 | |             .map_err(|layout| { err_inval!(Layout(layout)) })
    | |_____________________________________________________________^ expected struct `rustc::mir::interpret::InterpErrorInfo`, found enum `rustc::mir::interpret::InterpError`
    |
    = note: expected type `std::result::Result<rustc_target::abi::TyLayout<'_, &rustc::ty::TyS<'_>>, rustc::mir::interpret::InterpErrorInfo<'_>>`
               found type `std::result::Result<rustc_target::abi::TyLayout<'_, &rustc::ty::TyS<'_>>, rustc::mir::interpret::InterpError<'_>>`
```

Looks like we need the `into`?

Reply:
There's no `?` anywhere so indeed we need a manual `into()`.